### PR TITLE
Added confirmation popup on release notes

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

Show Prerequisite modal before upgrade to new version, So that it is upfront visible to the user and don't get overlooked
* Show a prerequisite box separately below the install button
* Show a prerequisite modal on click of install button
* Proceed further only when user provides confirmation

Fixes # [(issue)](https://github.com/devtron-labs/dashboard/issues/450)
Screenshot 2022-07-26 at 6 04 57 PM
Screenshot 2022-07-26 at 6 05 05 PM
Screenshot 2022-07-26 at 6 05 20 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Manually tested

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 1]
```diff
@@ -24,8 +24,10 @@ import { ReactComponent as File } from '../../../assets/icons/ic-file-text.svg'
 import { ReactComponent as Chat } from '../../../assets/icons/ic-chat-circle-dots.svg'
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:24]: Can remove `ReactComponent as File` line number 23 not using anywhere

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 2]
```diff
@@ -24,8 +24,10 @@ import { ReactComponent as File } from '../../../assets/icons/ic-file-text.svg'
...
 import { ReactComponent as Info } from '../../../assets/icons/info-filled.svg'
 import { ReactComponent as Warning } from '../../../assets/icons/ic-warning.svg'
+import { ReactComponent as Note } from '../../../assets/icons/ic-note.svg'
+import { ReactComponent as CloseIcon } from '../../../assets/icons/ic-close.svg'
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:29]: please remove the empty line (29) if not done purposely

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 3]
```diff
@@ -429,91 +431,214 @@ export const InstallationWrapper = ({
...
 }: InstallationWrapperType): JSX.Element => {
     const history: RouteComponentProps['history'] = useHistory()
     const location: RouteComponentProps['location'] = useLocation()
     const latestVersionAvailable = isLatestVersionAvailable(serverInfo?.currentVersion, upgradeVersion)
+    const [preRequisiteList, setPreRequisiteList] = useState<
+        { version: string; prerequisiteMessage: string; tagLink: string }[]
+    >([])
+    useEffect(() => {
+        if (releaseNotes) {
+            const _preRequisiteList = []
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:449]: Can we make a separate function for this and then call it in useEffect?

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:23]: Done

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 4]
```diff
@@ -429,91 +429,214 @@ export const InstallationWrapper = ({
...
+            for (let index = 0; index < releaseNotes.length; index++) {
+                const element = releaseNotes[index]
+                if (serverInfo && serverInfo.currentVersion === element.releaseName) {
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:450]: Can do this instead `serverInfo?.currentVersion`

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: done

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 5]
```diff
@@ -429,91 +429,214 @@ export const InstallationWrapper = ({
...
-                        (installationStatus !== ModuleStatus.HEALTHY ||
-                            (installationStatus === ModuleStatus.HEALTHY && latestVersionAvailable)) && (
-                            <button
-                                className="module-details__install-button cta flex mb-16"
-                                onClick={handleActionButtonClick}
+    const handlePrerequisiteCheckedChange = (): void => {
+        setPreRequisiteChecked(!preRequisiteChecked)
+    }
+
+    const renderConfirmationModal = (): JSX.Element | null => {
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:483]: Can we name it as  `renderPrerequisiteConfirmationModal`

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 6]
```diff
@@ -429,91 +429,214 @@ export const InstallationWrapper = ({
...
+        return (
+            <VisibleModal className="transition-effect">
+                <div className="modal__body upload-modal no-top-radius mt-0 p-0 w-600">
+                    <div className="flexbox content-space pl-20 pr-20 pt-16 pb-16 border-bottom">
+                        <div className="fw-6 fs-16 cn-9">
+                            {`Pre-requisites for update to ${upgradeVersion.toLowerCase()}`}
+                        </div>
+                        <CloseIcon
+                            className="pointer mt-2"
+                            onClick={() => setShowPreRequisiteConfirmationModal(false)}
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: done

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 5 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: done

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: done

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: done

**Review by shivani170 — APPROVED** [review]
Looks Good

**Pull request merged**